letsencrypt/tests/auth_handler_test.py

```diff
@@ -148,6 +148,42 @@ def test_perform_failure(self):
         self.assertRaises(
             errors.AuthorizationError, self.handler.get_authorizations, ["0"])
 
+    @mock.patch("acme.challenges.HTTP01Response.simple_verify")
+    @mock.patch("acme.challenges.TLSSNI01Response.simple_verify")
+    def test_simple_verify(self, mock_http01_verify, mock_tlssni01_verify):
+        from acme import jose
+        from letsencrypt.tests import test_util
+
+        key = jose.JWK.load(test_util.load_vector('rsa512_key.pem'))
+        http_01 = achallenges.KeyAuthorizationAnnotatedChallenge(
+            challb=acme_util.HTTP01_P, domain=b'localhost', account_key=key)
+        tls_sni_01 = achallenges.KeyAuthorizationAnnotatedChallenge(
+            challb=acme_util.TLSSNI01_P, domain=b'localhost', account_key=key)
+
+        dv_c_http = [http_01]
+        responses_http = gen_http01_resp(dv_c_http)
+        dv_c_tls = [tls_sni_01]
+        responses_tls = gen_tls_resp(dv_c_tls)
+
+        responses = responses_http + responses_tls
+
+        self.handler.dv_c = dv_c_http + dv_c_tls
+        self.mock_dv_auth.perform.side_effect = [responses]
```

Comment:
Ah, `self.mock_dv_auth.perform.side_effect` is set somewhere else, in `setUp`, before this point is reached. Thanks for clarification.